sys/src/libthread/BUILD

```diff
@@ -8,7 +8,7 @@ cc_library(
             "//amd64/include",
         ],
 	srcs = [
-		"amd64.c",
+		"$ARCH.c",
```

Comment:
well my question was more about whether we should keep the go functionality as defined [here](https://golang.org/src/os/env.go?s=378:435#L3) or be more opinionated. I'll change it to`${[A-Z]*}`